CRM/Report/Form/Activity.php

```diff
@@ -781,13 +781,13 @@ public static function formRule($fields, $files, $self) {
     return $errors;
   }
 
-  public function postProcess() {
-    //reset value of activity_date
-    if (!empty($this->_resetDateFilter)) {
-      $this->_formValues["activity_date_time_relative"] = NULL;
-    }
-    $this->beginPostProcess();
-    $activityContacts = CRM_Activity_BAO_ActivityContact::buildOptions('record_type_id', 'validate');
+  /**
+   * @param $applyLimit
+   *
+   * @return string
+   */
+  public function buildQuery($applyLimit = TRUE) {
+    $activityContacts = CRM_Core_OptionGroup::values('activity_contacts', FALSE, FALSE, FALSE, NULL, 'name');
```

Comment:
just in terms of preferred functions I think the original function is the preferred one. Although a still preferred & slightly shorter option is 
 $sourceID = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_ActivityContact', 'record_type_id', Activity Source');

I do really feel that we should have a \Civi interface for this since I think the case is being made that interacting via \Civi from extensions is supported - although I'm not sure how clear that is in the docs - will have to recheck

Reply:
Oops, my bad, I did a mistake merging code. In 44f817d4374, mattwire had replaced with: `$activityContacts = CRM_Activity_BAO_ActivityContact::buildOptions('record_type_id', 'validate');` in many places. I think we should stick to that?